Model/Invoice/StatusUpdater.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * Youama_OTP
+ *
+ * @author  David Belicza <87.bdavid@gmail.com>
+ * @license David Belicza e.v. (http://youama.hu)
+ */
+
+declare(strict_types=1);
+
+namespace Youama\OTP\Model\Invoice;
+
+use Magento\Sales\Api\Data\OrderStatusHistoryInterfaceFactory;
+use Magento\Sales\Api\OrderRepositoryInterface;
+use Magento\Sales\Api\InvoiceRepositoryInterface;
+use Magento\Sales\Model\Order;
+use Magento\Sales\Model\Order\Email\Sender\InvoiceSender;
+use Magento\Sales\Model\Service\InvoiceService;
+use Magento\Framework\DB\Transaction;
+use Magento\Sales\Model\Order\Status\History;
+use Youama\OTP\Helper\Config;
+
+/**
+ * Class StatusUpdater
+ *
+ * This class manage to create an invoice for the order if all the items are virtual and the auto-invoicing option is
+ * enabled in the admin
+ * History and Comment changes.
+ *
+ * @package Youama\OTP\Model\Order
+ */
+class StatusUpdater
+{
+    /**
+     * @var OrderRepositoryInterface
+     */
+    private $orderRepository;
+
+    /**
+     * @var OrderStatusHistoryInterfaceFactory
+     */
+    private $orderStatusHistoryInterfaceFactory;
+
+    /**
+     * @var Config
+     */
+    private $configHelper;
+
+    /**
+     * @var InvoiceService
+     */
+    protected $invoiceService;
```

Comment:
These fields don't have to be protected, they can be private according to SOLID principles. Same in Order's StatusUpdater.

Reply:
Fixed